spec/stack_master/commands/apply_spec.rb

```diff
@@ -122,24 +139,29 @@ def apply
   context 'the stack does not exist' do
     let(:stack) { nil }
 
-    it 'calls the create stack API method' do
+    it 'creates a change set for a new stack' do
       apply
-      expect(cf).to have_received(:create_stack).with(
+      expect(StackMaster::ChangeSet).to have_received(:create).with(
         stack_name: stack_name,
         template_body: proposed_stack.template_body,
         parameters: [
           { parameter_key: 'param_1', parameter_value: 'hello' }
         ],
         tags: [
-          {
-            key: 'environment',
-            value: 'production'
-          }],
+          { key: 'environment', value: 'production' }
+        ],
         capabilities: ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
         role_arn: role_arn,
         notification_arns: [notification_arn],
-        stack_policy_body: stack_policy_body,
-        on_failure: 'ROLLBACK'
```

Comment:
Sounds reasonable. I've added the original specs for the case when `on_failure` option is set to the below context block.